packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -824,10 +824,7 @@ export function diffProperties(
         // inserted already.
       }
     } else if (propKey === CHILDREN) {
-      if (
-        lastProp !== nextProp &&
```

Comment:
This change seems nice, since we already check for this above

https://github.com/facebook/react/blob/c9e59f0a311c7aada51e4a5dd76dc01ea454259c/packages/react-dom/src/client/ReactDOMComponent.js#L767